api/app/utils/polygonize.py

```diff
@@ -30,12 +30,13 @@ def polygonize_in_memory(geotiff_filename, layer, field) -> ogr.Layer:
     source: gdal.Dataset = gdal.Open(geotiff_filename, gdal.GA_ReadOnly)
 
     source_band = source.GetRasterBand(1)
+    nodata_value = source_band.GetNoDataValue()
     source_data = source.ReadAsArray()
     # https://gdal.org/api/python/osgeo.osr.html#osgeo.osr.SpatialReference
     spatial_reference: osr.SpatialReference = source.GetSpatialRef()
 
     # generate mask data
-    mask_data = np.where(source_data == 0, False, True)
+    mask_data = np.where(source_data == nodata_value, False, True)
```

Comment:
Could anything bad happen here if nodata_value is None? Or is it possible for a raster not to have a no_data value set?

Reply:
Good thought. I don't think anything bad would happen but I should check. It is possible for a raster to not have a NoData value, in which case `GetNoDataValue()` would return None

Reply:
Tested, no issues. Gets uploaded the same way as it did previously since there were never any 0 values in either of the fuel grids

Reply:
Nice. Also looks like snow processing (which also relies on polygonization) worked with the exception of a data problem at the source (NSIDC).